### PR TITLE
mplayer: fix improper autodetection of libbs2b availability

### PR DESCRIPTION
**Pull request by smaarn**
Issue: When trying to build `mplayer` formula, it fails with the following error:

```
libaf/af_bs2b.c:24:10: fatal error: 'bs2b.h' file not found
#include <bs2b.h>
         ^~~~~~~~
```

I'm using Mac OS X 10.13 (guilty as charged) with XCode 10.1.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

EDIT: added the issue description

### CODE REVIEW

**Comment by carlocab** [comment]
What does adding this dependency do? Does the `configure` script find the dependency without additional configuration?

It also seems that `libbs2b` pulls in its own dependency tree... `libogg`, `flac`, `libvorbis`, etc. Can you check what the impact of these additional dependencies (i.e. `libbs2b` + its extra dependencies that aren't already in `mplayer`'s existing dependency tree) on the install size of `mplayer`?

File: Formula/mplayer.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class Mplayer < Formula
   homepage "https://mplayerhq.hu/"
   url "https://mplayerhq.hu/MPlayer/releases/MPlayer-1.4.tar.xz"
   sha256 "82596ed558478d28248c7bc3828eb09e6948c099bbd76bb7ee745a0e3275b548"
```
Comment (carlocab) [inline Formula/mplayer.rb:5]:
```suggestion
  sha256 "82596ed558478d28248c7bc3828eb09e6948c099bbd76bb7ee745a0e3275b548"
  license all_of: ["GPL-2.0-only", "GPL-2.0-or-later"]
```
From `Copyright`:
```
Most files in MPlayer are licensed under the GNU General Public License,
version 2 with an explicit "or later" clause or a compatible license. An
exception is the file libmpdemux/demux_ty_osd.c, which is version 2 only.
Thus MPlayer as a whole can be redistributed under the terms of the GNU
General Public License, version 2, see the LICENSE file for details.
```

**Comment by smaarn** [comment]
> What does adding this dependency do? Does the `configure` script find the dependency without additional configuration?

Now that you ask, it seems that it does, while checking the logs for more details I found this line:

```
clang -MMD -MP -D_ISOC99_SOURCE -I. -Iffmpeg -O2 -march=corei7-avx -pipe -fno-asynchronous-unwind-tables -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -mdynamic-no-pic  -fpie -DPIC  -I/usr/X11/include -I/usr/local/Cellar/libcaca/0.99b19_3/include/ -I/usr/local/opt/freetype/include/freetype2 -DZLIB_CONST -I/usr/local/include/bs2b -c -o libaf/af_bs2b.o libaf/af_bs2b.c
```

This seems to indicate that the include is indeed found by the configure phase. What I cannot understand, though, is why I need to add the dependency to have the build actually pass :(

> 
> It also seems that `libbs2b` pulls in its own dependency tree... `libogg`, `flac`, `libvorbis`, etc. Can you check what the impact of these additional dependencies (i.e. `libbs2b` + its extra dependencies that aren't already in `mplayer`'s existing dependency tree) on the install size of `mplayer`?

Unfortunately I can't tell the "old" size of mplayer since my install got overwritten by my fix :(.

I'll try to check this.

**Comment by carlocab** [comment]
>Unfortunately I can't tell the "old" size of mplayer

It's not just the size of mplayer, but also the size of libbs2b's dependency tree, which you don't need your old installation to find out.

Did you have libbs2b installed when your build failed? If you did, it might be that this is just opportunistic linkage. If that's the case, then the correct solution would be to find a way to stop it with a configure flag (e.g. `--disable-bs2b` or `--without-bs2b` or similar) rather than adding it as a dependency.

**Comment by carlocab** [comment]
From `./configure -h`:

    --disable-libbs2b         disable libbs2b audio filter support [autodetect]

It's probably better to add this configure flag rather than to add a dependency for something you don't find useful.

**Comment by smaarn** [comment]
> From `./configure -h`:
> 
> ```
> --disable-libbs2b         disable libbs2b audio filter support [autodetect]
> ```
> 
> It's probably better to add this configure flag rather than to add a dependency for something you don't find useful.

Technically I have no position regarding this dependency, meaning I have no strong opinion here. As a matter of fact I'm not depending on mplayer directly, as far as I recall. 
But, indeed, I actually had installed the library and the error seemed to be related to the fact that autodetection didn't really integrate well with symlink.
I'll add the disable flag instead and see if it works out.

**Comment by carlocab** [comment]
>Technically I have no position regarding this dependency, meaning I have no strong opinion here.

I think that's enough for us to say you don't find it useful. Unless an optional dependency enables functionality you're looking for, then we shouldn't add it.

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/mplayer.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@ class Mplayer < Formula
...
+  license all_of: ["GPL-2.0-only", "GPL-2.0-or-later"]
```
Comment (SMillerDev) [inline Formula/mplayer.rb:6]: How can it be both only GPL2 but also later?

Comment (carlocab) [reply Formula/mplayer.rb:6]: From `Copyright`:
```
Most files in MPlayer are licensed under the GNU General Public License,
version 2 with an explicit "or later" clause or a compatible license. An
exception is the file libmpdemux/demux_ty_osd.c, which is version 2 only.
Thus MPlayer as a whole can be redistributed under the terms of the GNU
General Public License, version 2, see the LICENSE file for details.
```
Just reporting what's in the headers

Comment (smaarn) [reply Formula/mplayer.rb:6]: Frol what it says in the header it should actually be one or the other. I don't know what we're doing of the mentioned file, though. We may indeed have to select the -only option.

**Review by carlocab — APPROVED** [review]
Thanks, @smaarn.

**Pull request closed without merging**